### ISSUE TITLE
Add Create/Read/Update/Delete Product Image

### ISSUE THREAD

**Issue by sudomake** [getconversio/go-shopify#47]

**Comment by sudomake** [comment]
It looks as though there are two ways to POST an image file:

- Source URL
- Encoded Image + File Name

Is it acceptable to have two methods and types for Create:

`Image.Create(Image)`
`Image.CreateFromAttachment(ImageAttachment)`

**Comment by dlebech** [comment]
@sudomake good question. It looks like the structure of the Image is the same except for the `src` or `attachment` fields so I think it would make sense to use the same structure and have both fields in the same structure. You could then use `omitempty` in the json definition which should ensure that only the non-empty field is sent to Shopify, e.g.:
```
type Image struct {
  // ... other fields
  Src        string `json:"src,omitempty"`
  Attachment string `json:"attachment,omitempty"`
  // ... other fields
}
```
This way, I think it would be enough with a single `Create` function. What do you think?

**Comment by sudomake** [comment]
Sounds good.  I wonder what Shopify will do if src, attachment, and filename are all populated and valid.

Edit: No error was thrown by Shopify.  It took the attachment and ignored the src.  It accepted the attachment even when no filename was given.

**Comment by dlebech** [comment]
Thank you @sudomake ! I merged #49 now, so I'll close this issue :)
